FWCore/ParameterSet/test/test_catch2_ps.cc

```diff
@@ -0,0 +1,875 @@
+      CHECK(*ret == std::string_view("a\0\0b\0", 5));
+    }
+    SECTION("null separator, stuff after") {
+      auto ret = edm::decode_string_extent(std::string_view("a\0\0b\0)>", 7));
+      REQUIRE(ret);
+      CHECK(*ret == std::string_view("a\0\0b\0", 5));
+    }
+    SECTION("null start") {
+      auto ret = edm::decode_string_extent(std::string_view("\0\0ab\0", 5));
+      REQUIRE(ret);
+      CHECK(*ret == std::string_view("\0\0ab\0", 5));
+    }
+    SECTION("null start, stuff after") {
+      auto ret = edm::decode_string_extent(std::string_view("\0\0ab\0)>", 7));
+      REQUIRE(ret);
+      CHECK(*ret == std::string_view("\0\0ab\0", 5));
+    }
+    SECTION("null end") {
+      auto ret = edm::decode_string_extent(std::string_view("ab\0\0\0", 5));
+      REQUIRE(ret);
+      CHECK(*ret == std::string_view("ab\0\0\0", 5));
+    }
+    SECTION("null end, stuff after") {
+      auto ret = edm::decode_string_extent(std::string_view("ab\0\0\0)>", 7));
+      REQUIRE(ret);
+      CHECK(*ret == std::string_view("ab\0\0\0", 5));
+    }
+  }
+  SECTION("string") {
+    SECTION("simple") {
+      testbody<std::string>("");
+      testbody<std::string>("Hello there");
+      testbody<std::string>("123");
+    }
+    SECTION("escaped characters") { testbody<std::string>("This\nis\tsilly\n"); }
+    SECTION("special to PSet characters") { testbody<std::string>("{This,is`silly}"); }
+    SECTION("all characters") {
+      std::array<char, 256> allChars;
+      for (int i = 0; i < 256; ++i) {
+        allChars[i] = static_cast<char>(i);
+      }
+      std::string const all(allChars.begin(), allChars.end());
+      REQUIRE(all.size() == 256);
+      testbody<std::string>(all);
+    }
+    SECTION("lots of nulls") {
+      testbody<std::string>(std::string("\0", 1));
+      testbody<std::string>(std::string("\0\0", 2));
+      testbody<std::string>(std::string("\0\0\0", 3));
+      testbody<std::string>(std::string("\0\0\0\0", 4));
+    }
+    SECTION("null separator") { testbody<std::string>(std::string("a\0b", 3U)); }
+    SECTION("null begin") { testbody<std::string>(std::string("\0ab", 3U)); }
+    SECTION("null end") { testbody<std::string>(std::string("ab\0", 3U)); }
+    SECTION(":") { testbody<std::string>("ab:c"); }
+    SECTION("existsAs") {
+      std::string s = "some value";
+      edm::ParameterSet p1;
+      p1.addParameter<std::string>("s", s);
+      p1.registerIt();
+      REQUIRE(p1.existsAs<std::string>("s"));
+      REQUIRE(not p1.existsAs<std::string>("not_here"));
+    }
+  }
+  SECTION("vstring") {
+    SECTION("simple") {
+      std::vector<std::string> vs;
+      vs.push_back("one");
+      vs.push_back("1");
+      vs.push_back("TWO");
+      vs.push_back("three");
+      testbody<std::vector<std::string>>(vs);
+      edm::ParameterSet p1;
+      p1.addParameter<std::vector<std::string>>("vs", vs);
+      p1.registerIt();
+      std::vector<std::string> vs2 = p1.getParameter<std::vector<std::string>>("vs");
```

Comment:
Would it be useful to compare `vs2` against something? (same question for other similar tests below)

Reply:
`testbody` does doe such a test. I'm not sure why the extra stuff is here, I was copying from the cppunit version.

Reply:
removed extra stuff after `testbody` as it was redundant.